test/jdk/java/util/zip/ZipFile/EndOfCenValidation.java

```diff
@@ -0,0 +1,228 @@
+import java.util.zip.ZipException;
+import java.util.zip.ZipFile;
+import java.util.zip.ZipOutputStream;
+
+import static org.testng.Assert.*;
+
+/**
+ * This test augments {@link TestTooManyEntries}. It creates sparse ZIPs where the
+ * the CEN size is inflated to the desired value. This helps this test run
+ * fast with much less resources.
+ *
+ * While the CEN in these files are zero-filled and the produced ZIPs are technically
+ * invalid, the CEN is never actually read by ZipFile since it does END
+ * record validation before reading the CEN.
+ */
+public class EndOfCenValidation {
+
+    // Zip files produced by this test
+    public static final Path CEN_TOO_LARGE_ZIP = Path.of("cen-size-too-large.zip");
+    public static final Path INVALID_CEN_SIZE = Path.of("invalid-zen-size.zip");
+    public static final Path BAD_CEN_OFFSET_ZIP = Path.of("bad-cen-offset.zip");
+    // Some ZipFile constants for manipulating the END header
+    private static final int ENDHDR = ZipFile.ENDHDR; // End of central directory record size
+    private static final int ENDSIZ = ZipFile.ENDSIZ; // Offset of CEN size field within ENDHDR
+    private static final int ENDOFF = ZipFile.ENDOFF; // Offset of CEN offset field within ENDHDR
+    // Maximum allowed CEN size allowed by ZipFile
+    private static int MAX_CEN_SIZE = Integer.MAX_VALUE - ENDHDR - 1;
+
+    // Expected message when CEN size does not match file size
+    private static final String INVALID_CEN_BAD_SIZE = "invalid END header (bad central directory size)";
+    // Expected message when CEN offset is too large
+    private static final String INVALID_CEN_BAD_OFFSET = "invalid END header (bad central directory offset)";
+    // Expected message when CEN size is too large
+    private static final String INVALID_CEN_SIZE_TOO_LARGE = "invalid END header (central directory size too large)";
+
+    // A valid ZIP file, used as a template
+    private byte[] zipBytes;
+
+    /**
+     * Create a valid ZIP file, used as a template
+     */
+    @BeforeTest
+    public void setup() throws IOException {
+        zipBytes = templateZip();
+    }
+
+    /**
+     * Delete big files after test, in case the file system did not support sparse files.
+     */
+    @AfterTest
+    public void cleanup() throws IOException {
+        Files.deleteIfExists(CEN_TOO_LARGE_ZIP);
+        Files.deleteIfExists(INVALID_CEN_SIZE);
+        Files.deleteIfExists(BAD_CEN_OFFSET_ZIP);
+    }
+
+    /**
+     * Validates that an end of central directory record with
+     * a CEN length exceeding {@link #MAX_CEN_SIZE} limit is rejected
+     */
+    @Test
+    public void shouldRejectTooLargeCenSize() throws IOException {
+        int size = MAX_CEN_SIZE + 1;
+
+        Path zip = zipWithModifiedEndRecord(size, true, 0, CEN_TOO_LARGE_ZIP);
+
+        ZipException ex = expectThrows(ZipException.class, () -> {
+            try (ZipFile zf = new ZipFile(zip.toFile())) {
+            }
+        });
```

Comment:
You can omit the try block

Reply:
I removed the try/catch block with a quivering lower lip :-)